src/osdp_pd.c

```diff
@@ -954,8 +958,8 @@ static int pd_receve_packet(struct osdp_pd *pd)
 	remaining = pd->rx_buf_len - len;
 	if (remaining) {
 		memmove(pd->rx_buf, pd->rx_buf + len, remaining);
-		pd->rx_buf_len = remaining;
 	}
+	pd->rx_buf_len = remaining;
```

Comment:
This is a separate bug fix :)

nice catch!

Reply:
If you don't mind can we keep this as part of this commit. It is required for this patch to work as expected.

Reply:
I just wanted to highlight that fix. Of course it can stay here.